### ISSUE TITLE
Support `matrix.rx[]` indexing

### ISSUE THREAD

**Issue by Wainberg** [rpy2/rpy2#1086]
It would be nice to support `matrix.rx[1, 2]` as an alternative to `matrix.rx(1, 2)`. This would also let you assign to a matrix using `matrix.rx[] = ...`.

**Comment by lgautier** [comment]
Yes. I once considered it but the subseting operator `[` in R can accept named arguments. The R documentation for `[` has"

```
Usage:

     x[i]
     x[i, j, ... , drop = TRUE]
     x[[i, exact = TRUE]]
     x[[i, j, ..., exact = TRUE]]
     x$name
     getElement(object, name)
     
     x[i] <- value
     x[i, j, ...] <- value
     x[[i]] <- value
     x$name <- value
```

Python does not allow it. This is a syntax error.

I checked this again and PEP-0472 could have resolved this but was ultimately rejected (https://peps.python.org/pep-0472/). Alternative methods propose a dictionary as an additional index to have named parameters. This could be a good solution.
